### ISSUE TITLE
o365 plugin is failing 2 unit tests after latest commits

### ISSUE THREAD

**Issue by rafaelmonterroza** [Microsoft/o365-moodle#275]
Hello! I downloaded the MOODLE_35_STABLE version of local_o365 in this repo because I'm interested in recent commits that contain bug fixes and found a couple tests failing:
local/o365/tests/sharepoint_test.php
local/o365/tests/coursegroups_test.php

The first one is being caused by the changes in the function local_o365\utils::is_o365_connected. Unfortunately, I'm still not able to find the reason for the second one to fail. I'll update this issue if I find something.

Thanks for your help.

**Comment by jamesmcq** [comment]
Thanks for the report @rafaelmonterroza - this will be resolved in the next release.

**Comment by rafaelmonterroza** [comment]
Hi @jamesmcq, I missed your answer here, sorry about that. I had time to review this failures and I believe I found the reasons behind them.

**local/o365/tests/sharepoint_test.php** failure:
This was caused because the function that verifies the token for a specific user no longer takes in consideration the data stored in local/o365. To fix this, I inserted a valid record in 'auth_oidc_token' table, after that the test passes successfully. 
This is the object I added in the test:
$tokenrec = (object) [
            'oidcuniqid' => 'user1',
            'resource' => 'https://graph.windows.net',
            'username' => $user1->username,
            'userid' => $user1->id,
            'scope' => 'User.Read',
            'authcode' => '000',
            'token' => '111',
            'refreshtoken' => '222',
            'idtoken' => '333',
            'expiry' => time() + 9999,
        ];

**local/o365/tests/coursegroups_test.php** failure:
This one fails because there are some ->apicall() that were replaced by ->betaapicall(). If I replace some of the URLs fields in $expectedrequests[], the test passes successfully.
I replaced in the Add and Remove requests https://graph.microsoft.com/v1.0/ 
with https://graph.microsoft.com/beta/

Now I'm experiencing something weird with the user sync task. But I'm not sure if that affects the correct functioning or if it is a expected behavior. I'll report it later.
